pkg/controller/ingress-shim/sync.go

```diff
@@ -160,6 +160,8 @@ func (c *controller) buildCertificates(ctx context.Context, ing *extv1beta1.Ingr
 			return nil, nil, err
 		}
 
+		c.setCommonName(crt, ing, tls)
```

Comment:
This relates to line 193, but I can't leave a comment on a line you didn't change..
I'm not sure if I understand the second question in your "Special notes for your reviewer" correctly, but on line 193 the commonName of an older cert that needs updating is always set to "", so it's not set at all as far as I can tell.

Reply:
Okay good catch... that line has to go

Reply:
Maybe I'm missing something, but it seems to me when updating an older cert the commonName is stilll set to ""?

Reply:
Will be updated by
```go
func certNeedsUpdate(a, b *cmapi.Certificate) bool {
 [..]	

	if a.Spec.CommonName != b.Spec.CommonName {
		return true
	}
```